File: java/com/countrygamer/arcanacraft/common/item/ItemArcana.java
package com.countrygamer.arcanacraft.common.item;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import net.minecraftforge.common.util.ForgeDirection;
import net.minecraftforge.fluids.FluidStack;

import com.countrygamer.arcanacraft.common.ACOptions;
import com.countrygamer.arcanacraft.common.ArcanaCraft;
import com.countrygamer.arcanacraft.common.block.ACBlocks;
import com.countrygamer.arcanacraft.common.extended.ExtendedArcanePlayer;
import com.countrygamer.countrygamercore.base.common.tile.TileEntityTankBase;
import com.countrygamer.countrygamercore.base.extended.ExtendedEntity;
import com.countrygamer.countrygamercore.common.item.ItemBase;

public class ItemArcana extends ItemBase {
	
	boolean debug = false;
	
	public ItemArcana(String modid, String name) {
		super(modid, name);
		this.setMaxStackSize(1);
	}
	
	@Override
	public ItemStack onItemRightClick(ItemStack itemStack, World world, EntityPlayer player) {
		ExtendedArcanePlayer arcanePlayer = (ExtendedArcanePlayer) ExtendedEntity.getExtended(
				player, ExtendedArcanePlayer.class);
		if (arcanePlayer == null) return itemStack;
		
		if (!arcanePlayer.isPlayerArcaic()) arcanePlayer.setArcaic(true);
		
		if (debug) {
			
		}
		else {
			int id = player.isSneaking() ? ACOptions.arcanaGuiReset : ACOptions.arcanaGui;
			player.openGui(ArcanaCraft.pluginID, id, world, (int) player.posX, (int) player.posY,
					(int) player.posZ);
		}
		
		return itemStack;
	}
	
	@Override
	public boolean onItemUse(ItemStack itemStack, EntityPlayer player, World world, int x, int y,
			int z, int side, float par8, float par9, float par10) {
		if (debug) {
			TileEntity tileEntity = world.getTileEntity(x, y, z);
			if (tileEntity != null && tileEntity instanceof TileEntityTankBase) {
				((TileEntityTankBase) tileEntity).fill(ForgeDirection.UNKNOWN, new FluidStack(
						ACBlocks.liquidManus, 1000), true);
			}
			return true;
		}
		return false;
	}
	
}

File: src/main/scala/com/countrygamer/arcanacraft/common/item/arcana/ItemArcana.scala
package com.countrygamer.arcanacraft.common.item.arcana

import com.countrygamer.arcanacraft.common.{ACOptions, ArcanaCraft}
import com.countrygamer.arcanacraft.common.extended.ArcanePlayer
import com.countrygamer.arcanacraft.common.item.ItemArcanaCraft
import com.countrygamer.cgo.wrapper.common.extended.ExtendedEntityHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.util.MathHelper
import net.minecraft.world.World

/**
 *
 *
 * @author CountryGamer
 */
class ItemArcana(name: String) extends ItemArcanaCraft(name) {

	// Default Constructor
	{

	}

	// End Constructor
	/**
	 * Called whenever this item is equipped and the right mouse button is pressed.
	 *
	 * @param itemStack
	 * @param world
	 * @param player
	 * @return
	 */
	override def onItemRightClick(itemStack: ItemStack, world: World,
			player: EntityPlayer): ItemStack = {

		val arcanePlayer: ArcanePlayer = ExtendedEntityHandler
				.getExtended(player, classOf[ArcanePlayer]).asInstanceOf[ArcanePlayer]

		if (!arcanePlayer.isArcaic()) {
			arcanePlayer.setArcaic(value = true)
		}

		player.openGui(ArcanaCraft, ACOptions.guiArcana, world,
			MathHelper.floor_double(player.posX),
			MathHelper.floor_double(player.posY),
			MathHelper.floor_double(player.posZ)
		)

		itemStack
	}

}
